gourmet/defaults/defaults.py

```diff
@@ -14,7 +14,8 @@
 
 # Windows locales are named differently, e.g. German_Austria instead of de_AT
 # Fortunately, we can find the POSIX-like type using a different method.
-elif os.name == 'nt':
+# sys.platform is the correct check per mypy convention (https://mypy.readthedocs.io/en/stable/common_issues.html?highlight=platform#python-version-and-system-platform-checks)
+elif sys.platform == "win32":
```

Comment:
Should this be in a straight if/else with the if block above? If you try to run it on a system where both conditions are False (which is admittedly unlikely), it looks like it will fail with a confusing NameError.

Reply:
well, the question is difficult. https://docs.python.org/3/library/os.html#os.name states that there's `posix`, `nt`, and `java`. Now I don't know what `java` means (Jython, somethingsomething embedded?). They do hint at `sys.platform`. But the locale is surely only POSIX-dependent, right?

`sys.platform` (https://docs.python.org/3/library/sys.html#sys.platform) is more complex, as there are numerous switches for different flavors of UNIX.

As I see it, the case where this fails is this strange `java` platform. And as long as we don't know what this is, I am fine with failing at this point.

Maybe we can put an else in there, as we don't know how to find the locale. But as you said, this case is unlikely.

Reply:
Yup, `java` is Jython, IIRC. At least at one point, Jython identified itself in `os.name` and IronPython in `sys.platform`, just to be tricky. But AFAIK neither supports Python 3, so I don't think we need to worry about them.

I might do `if sys.platform == 'win32'` and then a bare `else` for the POSIX case. If someone tries on a strange platform, a failure on `setlocale` is probably pretty obvious.